pkg/cloud/cluster.go

```diff
@@ -69,13 +69,16 @@ func (c *client) GetOrCreateCluster(csCluster *infrav1.CloudStackCluster) (retEr
 
 	// If provided, translate Domain name to Domain ID.
 	if csCluster.Spec.Domain != "" {
```

Comment:
Isn't this `Domain.Name` now?

Reply:
No, that'd break the API, right?  Our goal is to keep it the same as it is today.